quic/s2n-quic-transport/src/connection/connection_impl.rs

```diff
@@ -1445,6 +1439,12 @@ impl<Config: endpoint::Config> connection::Trait for ConnectionImpl<Config> {
                 packet_interceptor,
             )?;
 
+            // there are no pending ACKs to process so wipe the structure to avoid
+            // an unnecessary connection wakeup
+            if space.pending_ack_ranges.is_empty() {
+                space.pending_ack_ranges.wipe();
+            }
```

Comment:
why is this necessary? I thought the connection would only have ack interest if the ack ranges were non-empty?

Reply:
Wipe, cleans the current_active_path. We dont need to do this but its better for sanity checks that the current path is set to None.

Since there is no interest, the connection wont be woken up and wipe wont be called otherwise.

Reply:
renamed `wipe` -> `reset` because `reset` functionality is used in another place and I want to avoid extra methods on PendingAckRanges. Also clarified comments around the `is_empty`/`reset` check

`reset_aggregate_info` : reset info
`reset` : reset_aggregate_info + reset current_active_path

Reply:
https://github.com/aws/s2n-quic/pull/1298/commits/6d9ad20176a118f08f8f86473b1a2522db8a039c